.github/scripts/compile.sh

```diff
@@ -42,13 +42,19 @@ tar -zxf qpid-proton.tar.gz -C qpid-proton-src --strip-components 1
 do_patch "patches/proton" qpid-proton-src
 
 cd proton_build
-cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DBUILD_BINDINGS=python -DCMAKE_INSTALL_PREFIX=/usr -DSYSINSTALL_PYTHON=ON -DSSL_IMPL=openssl $WORKING/qpid-proton-src/ \
-    && make \
-    && make DESTDIR=$WORKING/proton_install install \
+cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+  -DCMAKE_C_FLAGS='-fno-semantic-interposition' -DENABLE_LINKTIME_OPTIMIZATION=ON \
+  -DSSL_IMPL=openssl -DBUILD_STATIC_LIBS=ON -DBUILD_BINDINGS=python -DSYSINSTALL_PYTHON=ON \
+  -DBUILD_EXAMPLES=OFF -DBUILD_TESTING=OFF \
+  -DCMAKE_INSTALL_PREFIX=/usr $WORKING/qpid-proton-src/ \
+    && VERBOSE=1 make DESTDIR=$WORKING/proton_install install \
     && tar -z -C $WORKING/proton_install -cf /qpid-proton-image.tar.gz usr \
-    && make install
+    && VERBOSE=1 make install
 cd $WORKING/build
-cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DUSE_LIBWEBSOCKETS=ON -DCMAKE_INSTALL_PREFIX=/usr $WORKING/ \
-    && make  \
-    && make DESTDIR=$WORKING/staging/ install \
+cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+  -DCMAKE_C_FLAGS='-fno-semantic-interposition' -DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON \
```

Comment:
This might be premature to consider for production build, but I want to wave it around for general consideration. Works for me.

Reply:
`-fno-semantic-interposition` afaik precludes doing `LD_PRELOAD=/usr/lib64/libjemalloc.so`, which is another performance-enhancing option (with significant performance impact on my laptop)... Yeah, probably not touch it until the best combination of these options is settled.

Reply:
> `-fno-semantic-interposition` afaik precludes doing `LD_PRELOAD=/usr/lib64/libjemalloc.so`

That's nonsense, the option works completely differently from what I thought. In any case, I saw no improvement from `-fno-semantic-interposition`, so I removed it from PR.